### PR TITLE
Add New Xray Scan Logic

### PR DESCRIPTION
**Pull request by talarian1**
Adds support for an Xray scan with context: JFrog platform project name or list of watches.
Adds support for authentication with an access token.
Adds UI field of "fixed version"

### CODE REVIEW

File: JFrogVSExtension/UI/Panel/DataService.cs [hunk 1]
```diff
@@ -190,11 +182,33 @@ private void InitializeComponent()
             this.components = new Dictionary<string, Component>();
             this.componentsCache = new HashSet<Components>();
         }
+        private Artifacts ParseCliAuditJson(string scanResults)
```
Comment (eyalbe4) [inline JFrogVSExtension/UI/Panel/DataService.cs:185]: Let's add a new-line before this method.

File: JFrogVSExtension/UI/Panel/DataService.cs [hunk 2]
```diff
@@ -190,11 +182,33 @@ private void InitializeComponent()
...
+                    var id = entry.Key.Substring(entry.Key.IndexOf("://"));
+                    var artifact = new Artifact();
+                    var issue = new Issue(securityIssue.Severity, securityIssue.Summery, "", entry.Value.ImpcatPAths[0][0].ComponentId, entry.Value.FixedVersions.ToString());
+                    artifact.general = new GeneralInfo()
+                    {
+                        Name = id,
+                        ComponentId = entry.Value.ImpcatPAths[0][0].ComponentId,
+                    };
+                    artifact.Issues.Add(issue);
+                artifacts.artifacts.Add(artifact);
```
Comment (eyalbe4) [inline JFrogVSExtension/UI/Panel/DataService.cs:201]: Line is misaligned.

File: JFrogVSExtension/UI/Panel/DataService.cs [hunk 3]
```diff
@@ -190,11 +182,33 @@ private void InitializeComponent()
...
+            }
+            return artifacts;
+        }
     }
 
     public enum RefreshType
     {
         Hard, Soft, None
     }
+
```
Comment (eyalbe4) [inline JFrogVSExtension/UI/Panel/DataService.cs:212]: Redundant new-line.

File: JFrogVSExtension/Utils/HttpClient/HttpUtils.cs [hunk 4]
```diff
@@ -13,28 +13,39 @@ namespace JFrogVSExtension.HttpClient
...
+                return XrayUtil.LoadServerStatus(resultFromXray);
             }
             catch (Exception e)
             {
                 throw new IOException(e.Message, e);
             }
         }
-
-        public static async Task<XrayVersion> GetVersionAsync()
+        public static async Task GetArtifactoryPingAsync()
```
Comment (eyalbe4) [inline JFrogVSExtension/Utils/HttpClient/HttpUtils.cs:31]: Let's add new-lines between all the methods in this file.

File: JFrogVSExtension/Utils/HttpClient/XrayHttpClient.cs [hunk 5]
```diff
@@ -8,27 +8,41 @@
 
 namespace JFrogVSExtension.HttpClient
 {
-    class XrayHttpClient
+    class JfrogHttpClient
     {
```
Comment (eyalbe4) [inline JFrogVSExtension/Utils/HttpClient/XrayHttpClient.cs:13]: Redundant new-line.

File: JFrogVSExtension/Utils/HttpClient/XrayHttpClient.cs [hunk 6]
```diff
@@ -8,27 +8,41 @@
...
+        private string artifactoryUrl = "";
+        private string user = "";
+        private string password = "";
+        private string userAgent = "jfrog-visual-studio-extension";
         private System.Net.Http.HttpClient httpClient = null;
-        public async Task<HttpResponseMessage> PerformGetRequestAsync(String usage)
+        private bool UseAccessToken { get => string.IsNullOrEmpty(user); }
+
+
+
```
Comment (eyalbe4) [inline JFrogVSExtension/Utils/HttpClient/XrayHttpClient.cs:23]: Redundant new-lines.

File: JFrogVSExtension/Utils/HttpClient/XrayHttpClient.cs [hunk 7]
```diff
@@ -81,24 +93,35 @@ private void InitClient()
...
             {
-                
-                //Create a query
                 System.Net.Http.HttpClient client = new System.Net.Http.HttpClient();
-                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes(
-                                                                    string.Format("{0}:{1}", user, password))));
+                if (!UseAccessToken)
+                {
+                    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", Convert.ToBase64String(Encoding.ASCII.GetBytes(
+                                                                        string.Format("{0}:{1}", user, password))));
```
Comment (eyalbe4) [inline JFrogVSExtension/Utils/HttpClient/XrayHttpClient.cs:100]: Let's fix this line's alignment.

File: JFrogVSExtension/Utils/ScanManager/ScanManager.cs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        #region Private fields
+        private const string CliServerId = "jfrog-visual-studio-extension";
+        private readonly Dictionary<string, string> cliEnv = new Dictionary<string, string> { { "CI", "TRUE" } };
+        private string project;
+        private string watches;
+        private static ScanManager instance = null;
+        private ScanManager() { }
+        #endregion
+        #region Public Properties
+        public static ScanManager Instance
```
Comment (eyalbe4) [inline JFrogVSExtension/Utils/ScanManager/ScanManager.cs:24]: Let's add new-line between the methods in this class.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: JFrogVSExtension/UI/Panel/DataService.cs [hunk 9]
```diff
@@ -186,34 +186,58 @@ private void InitializeComponent()
...
-                foreach (var entry in securityIssue.Components) {
-                    var artifactId = GetIdWithoutPackagePrefix(entry.Key);
-                    var directDependencyId = GetIdWithoutPackagePrefix(entry.Value.ImpactPaths[0][0].ComponentId);
-                    Artifact artifact;
-                    if (artifacts.ContainsKey(artifactId))
-                    {
-                        artifact = artifacts[artifactId];
-                    } else
+            foreach (var auditResult in auditResults) {
+                // Handel security issuses (violations and vulnerabilities)
```
Comment (RobiNino) [inline JFrogVSExtension/UI/Panel/DataService.cs:194]: Handel -> Handle.

File: JFrogVSExtension/UI/Panel/DataService.cs [hunk 10]
```diff
@@ -186,34 +186,58 @@ private void InitializeComponent()
...
+                        }
+                        var issueType = string.IsNullOrEmpty(securityIssue.IssueType) ? "security" : securityIssue.IssueType;
+                        var fixedVerions = entry.Value.FixedVersions != null ? string.Join(" ", entry.Value.FixedVersions) : "";
+                        var issue = new Issue(securityIssue.Severity, securityIssue.Summary, issueType, directDependencyId, fixedVerions);
+                        if (!artifact.Issues.Contains(issue))
+                        {
+                            artifact.Issues.Add(issue);
+                        }
+                    }
+                    // Handel licenses information
```
Comment (RobiNino) [inline JFrogVSExtension/UI/Panel/DataService.cs:221]: Handel -> Handle.

File: JFrogVSExtension/UI/Panel/DataService.cs [hunk 11]
```diff
@@ -186,34 +186,58 @@ private void InitializeComponent()
...
+                    foreach (var license in auditResult.Licenses)
                     {
-                        artifact = new Artifact
+                        foreach (var entry in license.Components)
                         {
-                            ArtifactId = artifactId,
-                        };
-                        artifacts.Add(artifactId, artifact);
+                            var artifactId = GetIdWithoutPackagePrefix(entry.Key);
+                            if (!artifacts.ContainsKey(artifactId))
```
Comment (RobiNino) [inline JFrogVSExtension/UI/Panel/DataService.cs:227]: Create a function to remove duplication with line 206

**Review by RobiNino — CHANGES_REQUESTED** [review]
Looks good!

**Pull request merged**